scene/resources/particles_material.cpp

```diff
@@ -260,7 +271,183 @@ void ParticlesMaterial::_update_shader() {
 	code += "	return x;\n";
 	code += "}\n";
 	code += "\n";
+	
+	//functions to create simplex noise, curl noise in 3D and 4D 
+	code += "\n\n";
+	code += "vec4 permute(vec4 x){return mod(((x*34.0)+1.0)*x, 289.0);}\n";
+
+	code += "vec4 taylorInvSqrt(vec4 r){return 1.79284291400159 - 0.85373472095314*r;}\n";
+
+	code += "float permute_float(float x){return floor(mod(((x*34.0)+1.0)*x, 289.0));}\n";
+
+	code += "float taylorInvSqrt_float(float r){return 1.79284291400159 - 0.85373472095314 * r;}\n";
```

Comment:
These four functions should be separated into multiple lines, not made into one-liners.